### ISSUE TITLE
[java] A false negative about InsecureCryptoIv

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3456]
**Affects PMD Version:6.42.0** 

**Rule:InsecureCryptoIv**

**Link:**
https://pmd.github.io/pmd-6.36.0/pmd_rules_java_security.html#insecurecryptoiv

**Description:**
Hi, I found a false negative when tried to use PMD to enhance code security. The minimal case is below. Our tool should report a warning at line 6, because secret key is originated from a constant string. This FN may lead to privacy information leakage.

**Code Sample demonstrating the issue:**

```java
public void func(SecretKeySpec key) {
    try {
        byte[] ivBytes, ivs;
        ivs = "hardcoded initial vector".getBytes();
        final IvParameterSpec iv = new IvParameterSpec(ivs); // should report a warning about this line
        final Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
        // .......
    } catch (Exception e) {
        e.printStackTrace();
    }
}
```

**Expected outcome: A warning**

PMD should report a violation at line 6, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Zustin** [comment]
@adangel  Hi, Andreas, could you please help check this issue? Thanks.

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0, although it will report on the previous line (hardcoded assignment to `ivs`).
